service/controller/resource/machinepoolconditions/conditionupgrading.go

```diff
@@ -0,0 +1,59 @@
+package machinepoolconditions
+
+import (
+	"context"
+
+	aeconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions"
+	"github.com/giantswarm/microerror"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
+
+	"github.com/giantswarm/azure-operator/v5/pkg/conditions"
+	"github.com/giantswarm/azure-operator/v5/pkg/upgrade"
+	"github.com/giantswarm/azure-operator/v5/service/controller/key"
+)
+
+func (r *Resource) ensureUpgradingCondition(ctx context.Context, machinePool *capiexp.MachinePool) error {
+	r.logDebug(ctx, "ensuring condition Upgrading")
+	var err error
+
+	// Let's make sure that the condition status is set to a supported value.
+	if conditions.IsUnexpected(machinePool, aeconditions.UpgradingCondition) {
+		return conditions.UnexpectedConditionStatusError(machinePool, aeconditions.UpgradingCondition)
+	}
+
+	// Set initial Upgrading condition status to false, since the MachinePool
+	// is just created.
+	if capiconditions.IsUnknown(machinePool, aeconditions.UpgradingCondition) {
+		err = conditions.MarkUpgradingNotStarted(machinePool)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		return nil
+	}
```

Comment:
Current logic for setting the `MachinePool` `Upgrading` condition in the first reconciliation loop - it just gets set to `False`. This should be good enough now, as every node pool is a new node pool (new clusters get new node pools from the start, legacy clusters get a first node pool which is also new, therefore no upgrades in the beginning), so this would practically initialize `Upgrading` condition for all node pools.

However, it could be too simple for other scenarios we talked about, e.g. restoring CRs where we would need to recalculate everything in the `Status`, including conditions.

Reply:
> However, it could be too simple for other scenarios we talked about, e.g. restoring CRs where we would need to recalculate everything in the Status, including conditions.

This is flaw in the implementation for the `Cluster` `Creating` and `Cluster` / `MachinePool` `Upgrading` conditions, while all other `*Ready` conditions are being recalculated all the time, based on the state of resources/components.

Reply:
>  This should be good enough now, as every node pool is a new node pool (new clusters get new node pools from the start, legacy clusters get a first node pool which is also new, therefore no upgrades in the beginning), so this would practically initialize Upgrading condition for all node pools.

We need to remember this in testing so that we test from a cluster that has been created with this operator version.